workspaces/topology/plugins/topology/src/components/LocalizedTimestamp.tsx

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+
+import { useLanguage } from '../hooks/useLanguage';
+
+interface LocalizedTimestampProps {
+  date: string | Date | undefined;
+  dateStyle?: 'full' | 'long' | 'medium' | 'short';
+  timeStyle?: 'full' | 'long' | 'medium' | 'short';
+}
+
+export const LocalizedTimestamp: React.FC<LocalizedTimestampProps> = ({
+  date,
+  dateStyle = 'medium',
+  timeStyle = 'short',
+}) => {
+  const language = useLanguage();
+
+  if (!date) {
+    return <span>-</span>;
+  }
+
+  const dateObj = typeof date === 'string' ? new Date(date) : date;
+
+  // Map language codes to proper locale codes
+  const getLocale = (lang: string): string => {
+    switch (lang) {
+      case 'fr':
+        return 'fr-FR';
+      case 'de':
+        return 'de-DE';
+      case 'it':
+        return 'it-IT';
+      case 'es':
+        return 'es-ES';
+      case 'en':
+      default:
+        return 'en-US';
+    }
+  };
+
```

Comment:
Are you sure that this is needed? It looks to me that toLocaleString works also with short locales:

<img width="506" height="272" alt="image" src="https://github.com/user-attachments/assets/2bcba8c6-a465-42cf-a579-9ddac31bfac8" />

<img width="749" height="217" alt="image" src="https://github.com/user-attachments/assets/d21e3d40-05ed-455d-97e0-c21343863eea" />